src/configure/model/templateModels.ts

```diff
@@ -1,26 +1,44 @@
-import { ExtendedPipelineTemplate } from "./Contracts";
 import { AzureConnectionType, ServiceConnectionType, TargetKind, TargetResourceType } from "./models";
 
+export enum TemplateType {
+    remote,
+    local
+}
 export interface PipelineTemplate {
-    path: string;
     label: string;
-    language: string;
+    templateWeight: number;
+    templateType: TemplateType;
     targetType: TargetResourceType;
     targetKind: TargetKind;
+    language: string;
+}
+
+export interface PipelineTemplateMetadata {
+    templateId: string;
+    workingDirectory: string;
+    templateWeight: number;
+    templateDescription: string;
+    attributes: TemplateAttributes;
+}
+
+export interface RemotePipelineTemplate extends PipelineTemplate {
```

Comment:
won't we able to get it from PipelineTemplate?
will replicating it in RemoteTemplate be the right way?

Reply:
We won't we able to get from PipelineTemplate. Yes replicate them to RemotTemplate, its right to have these properties there.

Reply:
added

Reply:
@anuragc617 I have not yet seen your PR for finding working directory, but I want to tell one thing - you should use the working directory property, which is returned along with template id from template service, for the purpose of figuring our working directory rather than using template attributes.
The reason is that template service has already computed which applicationSetting can get deployed using what templates. And that information it is exposing by returning the working directory property. So we should use this instead of trying ourselves to co-relate between applicationSettings and templates.

Reply:
cc @tejasd1990

Reply:
@bishal-pdMSFT template's working directory can match to multiple repositoryAnalysisSettings workingDirectory. So to narrow down more on repositoryAnalysisSetting, filtering should be done on templates deployTarget, buildTarget  and language attributes.